Add support for Ubuntu 14.04 and Nessus Manager/Agent

Vincent-- opened the pull request:
I'd like to use this module on Ubuntu 14.04 and also allow users to configure Nessus Manager and Nessus Agent.

This PR is empty for now, I'd just like to know your thoughts about that? Are you happy to extends the cover of your module or should I create my own module?

Regards,
Vincent.

Vincent-- committed:
Add support for Ubuntu 14.04

adamcrews commented:
I no longer have access to nessus licenses to test with and thus have little ability to contribute on the nessus side.  With that said, I have no problem helping with the puppet code, and I'd love to have the contributions here.

Vincent-- commented:
Ok, thanks. I'll push my changes when they are ready.
Are you ok to add dependencies like wget for instance (https://forge.puppet.com/maestrodev/wget)?

adamcrews commented:
I'm not opposed to adding dependencies.  For that specific functionality, I prefer puppet/archive over maestrodev/wget because most of the environments I work in already are using archive.  It's not a hard rule, just a preference, so feel free to use wget if you like that better.

Vincent-- committed:
Add issues url

Vincent-- committed:
Add puppet-archive dependencies

Vincent-- committed:
Increase puppet-archive version requirement

Vincent-- committed:
Lint code

Vincent-- committed:
Remove puppet-archive dependency

Tenable doesnt provide download links that can be automatised

Vincent-- committed:
Allow installation from a local package

Vincent-- commented:
Nessus doesn't seem to be available as a ppa and it require a manual validation in order to be downloaded from Tenable website.
Is it the same approach on Centos? I don't understand how the current version of the module can work without a local version of Nessus package available

Vincent-- commented on manifests/params.pp:
```diff
@@ -1,10 +1,63 @@
+# Rather, it only sets values for parameters to be consumed by child classes.
 class nessus::params {
 
-  $package_name    = 'Nessus'
```
Shouldn't the package name be 'nessus' instead of 'Nessus' (no uppercase)?

adamcrews replied:
IIRC, the rpm package is named specifically with an uppercase N as the package name.  This should be simple enough to handle as a selector for each OS.

Vincent-- committed:
Optimise nessus_activation_code fact

* Remove '\n' at the end of the value (so it can be used in tests)
* Use 'fix --list' and grep the output instead of using
  'fetch --code-in-use' as the format is easier to parse
* Fix the split pattern so it works with the new nessus client

Vincent-- committed:
Add nessus_manager_ip fact

(only available with nessus 6.0+)

Vincent-- committed:
Add nessus_manager_linked_once fact

(only available with nessus 6.0+)

Vincent-- committed:
Add nessus_manager_port fact

(only available with nessus 6.0+)

Vincent-- committed:
Add nessus_version fact

Vincent-- committed:
Optimise nessus facts

Replace shell command by corresponding ruby functions

Vincent-- committed:
Finalise nessus agent functionality

* Move variables validation into each class
* Use virtual_* variable to simplify code and manage both Nessus and
  Nessus agent with the same class
* Manage users directly from nessus class with a hash
* Reorder puppet anchor so service is now applied before config and we
  don't restart the service after using nessuscli as this doesnt seem to
  be recommended and this was breaking the plugin db, see
  "_qdb_open:/opt/nessus/var/nessus/plugins-desc.db: Invalid table of contents")
* Allow Nessus to be installed from local packages

Vincent-- commented:
Most of the code is now present. I've done quite a big refactoring in the end but there shouldn't be any breaking change on the existing code.

@adamcrews could you start to review my changes and let me know if it's ok for you? I'd like to work on the corresponding tests but Travis seems to be broken for now. I'll update the documentation until it is fixed.

adamcrews commented:
I'll start taking a detailed look at all this over the next few days since I will finally have a break at work.  Thank you for all the work on this.

adamcrews commented:
I just fixed the test framework.  The lint checks are failing, but everything else is currently passing.  Can you please rebase your changes so you get the latest test framework, and then your changes should all work.

Vincent-- committed:
Merge remote-tracking branch 'upstream/master' into ubuntu_support

Vincent-- commented:
Hi Adam,

I've tried to update my branch with your last changes, but there is 2 issues:
1 - I'm still using puppet 3.8 (I know it's bad, but I cannot upgrade all my envs for now), so I've no way to test the changes on puppet 4 and also I'd like this module to support puppet 3.x for some few months until I upgrade :-)
2 - I've never used travis, so I've no idea how to fix the current issue with the build :-(

Let me know what you want me to do about these 2 points?

Vincent-- committed:
Fix typo

nikkonix committed:
Update nessus agent version to 7.1.1
